tests/scripts/all.sh

```diff
@@ -769,6 +782,95 @@ component_build_default_make_gcc_and_cxx () {
     make TEST_CPP=1
 }
 
+component_test_submodule_cmake () {
+    # USE_CRYPTO_SUBMODULE: check that the build works with CMake
+    msg "build: cmake, full config + USE_CRYPTO_SUBMODULE, gcc+debug"
+    scripts/config.pl full # enables md4 and submodule doesn't enable md4
+    scripts/config.pl unset MBEDTLS_MEMORY_BACKTRACE # too slow for tests
+    CC=gcc cmake -D USE_CRYPTO_SUBMODULE=1 -D CMAKE_BUILD_TYPE=Debug .
+    make
+    msg "test: top-level libmbedcrypto wasn't built (USE_CRYPTO_SUBMODULE, cmake)"
+    if_build_succeeded not test -f library/libmbedcrypto.a
+    msg "test: libmbedcrypto symbols are from crypto files (USE_CRYPTO_SUBMODULE, cmake)"
+    if_build_succeeded objdump -g crypto/library/libmbedcrypto.a | grep -E 'crypto/library$' > /dev/null
+    msg "test: libmbedcrypto uses top-level config (USE_CRYPTO_SUBMODULE, cmake)"
+    if_build_succeeded objdump -g crypto/library/libmbedcrypto.a | grep 'md4.c' > /dev/null
```

Comment:
In addition to my remark above about `if_build_succeeded`: `grep 'md4\.c` or `grep -F 'md4.c'`. And I think you should actually use `grep '^md4\.c'`.

Reply:
Also pre-existing, hence IMO out of scope for this PR.

Reply:
Resolved here: https://github.com/ARMmbed/mbedtls/pull/2438